src/pages/dar_application/ResearchPurposeStatement.js

```diff
@@ -31,168 +30,315 @@ export default function ResearchPurposeStatement(props) {
   const genderValues = ['F', 'M'];
   const [gender, setGender] = useState(props.gender);
 
+  const alertBackgroundColor = '#f3494930';
+
+  //NOTE: inputs have both isEmpty and isNil checks
+  //currently values are initialized as emptry strings as a way to maintain controlled inputs in components
+  //however the inputs, when given a value, can either be a string (as seen with gender), or a boolean
+  //isEmpty will give a false negative with booleans and isNil will give a false positive with empty strings
+
   return(
     div({ className: 'col-lg-10 col-lg-offset-1 col-md-12 col-sm-12 col-xs-12' }, [
-      fieldset({ disabled: !isNil(darCode) }, [
+      fieldset({ disabled: !isEmpty(darCode) }, [
 
         h3({ className: 'rp-form-title access-color' }, ['3. Research Purpose Statement']),
 
         div({ className: 'form-group' }, [
-          div({ className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group' }, [
-            label({ className: 'control-label rp-title-question' },
-              ['3.1 In order to ensure appropriate review, please answer the questions below:'])
-          ]),
-          div({ className: 'row no-margin' }, [
-            span({
-              className: 'cancel-color required-field-error-span', isRendered: invalidPurpose && showValidationMessages,
-              style: { 'marginLeft': '15px' }
-            }, ['All fields are required'])
-          ]),
-          div({ className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group' }, [
-            label({ className: 'control-label rp-choice-questions' },
-              ['3.1.1 Will this data be used exclusively or partially for a commercial purpose?'])
+          div({className: 'row no-margin'}, [
+            div({ className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group row no-margin' }, [
+              label({ className: 'control-label rp-title-question' },
+                ['3.1 In order to ensure appropriate review, please answer the questions below:'])
+            ]),
           ]),
-          div({ className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group' }, [
-            YesNoRadioGroup({
-              value: forProfit, onChange: handleRadioChange, id: 'forProfit', name: 'forProfit',
-              required: true
-            })
+          div({className: 'radio-question-container row no-margin', style: {
+            backgroundColor: showValidationMessages && forProfit.toString().length === 0 ? alertBackgroundColor : 'inherit'
```

Comment:
I hit an error here:

1. Create DAR from selected datasets
2. Navigate to step 4
3. Follow validation error link to step 2.
4. Enter data in step 2 until it is valid
5. Navigate to step 3, get error:
```
TypeError: Cannot read property 'toString' of undefined
ResearchPurposeStatement
src/pages/dar_application/ResearchPurposeStatement.js:54
  51 |   ]),
  52 | ]),
  53 | div({className: 'radio-question-container row no-margin', style: {
> 54 |   backgroundColor: showValidationMessages && forProfit.toString().length === 0 ? alertBackgroundColor : 'inherit'
     | ^  55 | }}, [
  56 |   div({ className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group ' }, [
  57 |     label({ className: 'control-label rp-choice-questions' },
```

Reply:
I hit the same error if having completed step 2, then navigating to step 4, and follow the validation error link to step 3.

Reply:
Another thought on this, I think `forProfit` is unique in that it is in step 2 and in step 3.

Reply:
I can't seem to reproduce it on my end, but the toString check is prone to error if the props are not initialized on controller mount. I can add a variable check (```forProfit && forProfit.toString()...```) on the inputs to ensure that if it's undefined the toString check won't go through